File: src/test/java/bad/robot/radiate/ui/PropertyChangeListenerStub.java
package bad.robot.radiate.ui;

import java.beans.PropertyChangeEvent;
import java.beans.PropertyChangeListener;
import java.util.ArrayList;
import java.util.List;

class PropertyChangeListenerStub implements PropertyChangeListener {

    private final List<String> results = new ArrayList<>();

    @Override
    public void propertyChange(PropertyChangeEvent event) {
        results.add(event.toString());
    }

    public boolean contains(String result) {
        return results.contains(result);
    }

    public int size() {
        return results.size();
    }
}

File: src/test/scala/bad/robot/radiate/ui/PropertyChangeListenerStub.scala
package bad.robot.radiate.ui

import java.beans.{PropertyChangeEvent, PropertyChangeListener}

class PropertyChangeListenerScalaStub extends PropertyChangeListener {

  private val results = new scala.collection.mutable.MutableList[String]

  def propertyChange(event: PropertyChangeEvent) {
    results += event.toString
  }

  def contains(result: String) = {
    results.contains(result)
  }

  def size = results.size
}